### PR TITLE
[type-declarations] Open isset check in StrictArrayParamDimFetchRectorTest, move to last position in type-declaration level

### PR DESCRIPTION
**Pull request by TomasVotruba**
In one project I've noticed there is no `array` filled where obviously should be one. 
Single `isset` might be triggered on null and key, but maybe more useful cases are ignored then. Let's open up this a bit and test in the wild, if brings more value :+1:

### CODE REVIEW

**Review by samsonasik — CHANGES_REQUESTED** [review]
There is rule `StrictArrayParamDimFetchRector` rule for it, for `0` index without context, it can be `string`, which maybe not array on purpose, see

https://github.com/rectorphp/rector-src/blob/edadebb5211eafe3d33f3b283519dc283e674854/rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php#L173-L177

see https://3v4l.org/8h8nm

**Comment by TomasVotruba** [comment]
I could not find this one, thanks 

I see lot of missed array types just to avoid string access. Will make this condition less strict to improve auto fill 👍

**Comment by samsonasik** [comment]
Then that's need on very last on level set list with `Int_` key, as risky, or just not register in the set list, user can manually use this rule when needed.

Also, still needs `ParentClassMethodTypeOverrideGuard->hasParentClassMethod()` check still needed.

File: rules-tests/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector/Fixture/cover_isset_with_dim_fetch.php.inc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\StrictArrayParamDimFetchRector\Fixture;
+
+final class CoverIssetWithDimFetch
+{
+    public function run(array $param, $item): bool
```
Comment (staabm) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector/Fixture/cover_isset_with_dim_fetch.php.inc:29]: in this case $param could be anything which implements ArrayAccess not just the rare case of string-offset access. 

in case you "wrongly guess" the `array` type a call with a ArrayAccess object will produce a fatal error at runtime - see https://3v4l.org/csROV

Comment (TomasVotruba) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector/Fixture/cover_isset_with_dim_fetch.php.inc:29]: Indeed, it's a choice of practice over risk of few edge cases. We apply same logic to other rules too, to ease manual daunting work on project upgrades.

If your codebase uses any type in this case, exclude the rule.

Comment (TomasVotruba) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector/Fixture/cover_isset_with_dim_fetch.php.inc:29]: Indeed, it's a choice of practice over risk of few edge cases. We apply same logic to other rules too, to ease manual daunting work on project upgrades.

If your codebase uses any type in this case, exclude the rule.
